File: backend/app/schemas/social.py
"""收藏 & 评论 Schema"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


# ==================== Favorite ====================

class FavoriteCreate(BaseModel):
    """创建收藏"""
    target_type: str = Field(..., max_length=50, description="收藏目标类型: scenic_spot/hotel/restaurant")
    target_id: int = Field(..., description="收藏目标ID")


class FavoriteResponse(BaseModel):
    """收藏响应"""
    id: int
    user_id: int
    target_type: str
    target_id: int
    created_at: datetime

    model_config = {"from_attributes": True}


# ==================== Review ====================

class ReviewCreate(BaseModel):
    """创建评论"""
    target_type: str = Field(..., max_length=50, description="评论目标类型: scenic_spot/hotel/restaurant")
    target_id: int = Field(..., description="评论目标ID")
    content: str = Field(..., description="评论内容")
    score: Optional[float] = Field(None, ge=0, le=5, description="评分")


class ReviewUpdate(BaseModel):
    """更新评论"""
    content: Optional[str] = None
    score: Optional[float] = Field(None, ge=0, le=5)


class ReviewResponse(BaseModel):
    """评论响应"""
    id: int
    user_id: int
    target_type: str
    target_id: int
    content: str
    score: Optional[float] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/types/social.ts
/**
 * 社交互动相关类型定义（对应后端 app/schemas/social.py）
 */

// ==================== 通用 ====================

export type TargetType = 'scenic_spot' | 'hotel' | 'restaurant'

// ==================== Favorite ====================

export interface Favorite {
  id: number
  user_id?: number
  target_type: TargetType
  target_id: number
  created_at?: string
  target?: unknown  // 后端可能内联返回目标详情
}

export interface CreateFavoriteParams {
  target_type: TargetType
  target_id: number
}

// ==================== Review ====================

export interface ReviewUser {
  id?: number
  username?: string
  avatar?: string | null
}

export interface Review {
  id: number
  user_id?: number
  target_type: TargetType
  target_id: number
  content: string
  score: number | string  // 后端可能返回 number 或 string
  created_at?: string
  updated_at?: string
  user?: ReviewUser | null
}

export interface CreateReviewParams {
  target_type: TargetType
  target_id: number
  content: string
  score: number
}

export interface UpdateReviewParams {
  content?: string
  score?: number
}
